src/main/webapp/app/tutor-exercise-dashboard/tutor-exercise-dashboard.component.html

```diff
@@ -193,6 +197,9 @@ <h2 *ngIf="tutorParticipationStatus === TRAINED || tutorParticipationStatus ===
                             <td></td>
                             <td>{{ 'artemisApp.tutorExerciseDashboard.new' | translate }}</td>
                             <td></td>
+                            <td *ngIf="exercise.type === ExerciseType_TEXT">
+                                {{ textSubmissionLanguageTranslationKey(unassessedSubmission) | translate }}
```

Comment:
If you use a method invocation in a template like this, it will be called all the time (you can test this by setting a breakpoint at the method in the component).
Normally in angular you would rather use a component property for this (see also discussion of the method below).

Reply:
You could do this completely in the template, e.g. something like:
```
{{ 'artemisApp.tutorExerciseDashboard.languages.' + (submission.language ? submission.language : 'artemisApp.tutorExerciseDashboard.languages.UNKNOWN') | translate }}
```